### PR TITLE
feat(ui): Show first-time UX. Fixes #7160

### LINKED ISSUE #7160: In-app survey
We currently don't really have any way to connect to our end users. Most bugs are raised by operators - proxies for the end users. This seems to be reflected in our survey, where ops-engineer is well represented, but ML-engineer, not so much.

We could use two techniques.

(1)

Ask the user about their use case the first time they open the UI, and forward them to public docs helping them with that use case (capturing what they choose):

https://www.appcues.com/blog/how-turbotax-makes-a-dreadful-user-experience-a-delightful-one

image

(2) 

After two weeks of using the users interface, we should ask them how they are doing?

image

(3)

When they get a major/minor version (not patches), we should signpost the blog to point out the new version and its features.

```

            You've just upgraded to v3.2. Did you know about???

 * HTTP template for making fast HTTP request-
 * Multi-cluster for running workflows that span clusters/namespaces

[ Show me all the new features ] [ Dismiss ] [ Never show this again ] 
```

### PR DESCRIPTION
**Pull request by alexec**
Don't bother creating a PR until you've done this:

* [ ] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

# Screenshots

image

image

image

image

### CODE REVIEW

File: ui/src/app/modals/new-version-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import {Modal} from '../shared/components/modal/modal';
+
+require('./new-version-modal.scss');
+/**
+ * The intention of this modal is to encourage update of new features.
```
Comment (alexec) [inline ui/src/app/modals/new-version-modal.tsx:6]: "uptake"

**Comment by edlee2121** [comment]
Looks great!
Perhaps delete ETL since it is very similar to data processing.
Similarly, delete operational analytics since that is similar to anomaly detection for now.

**Comment by edlee2121** [comment]
It would be good to gather more references to help users with these use cases.

**Review by edlee2121 — APPROVED** [review]

**Comment by alexec** [comment]
> It would be good to gather more references to help users with these use cases: "We'll tell you how to do it."

not sure what you mean @edlee2121

**Comment by edlee2121** [comment]
Do we have enough documentation/solution guides for these use cases.

**Comment by alexec** [comment]
> Do we have enough documentation/solution guides for these use cases.

I struggled to find something substantial for each use case. Take a look at docs/use-cases

**Comment by alexec** [comment]
Let's chat about this tomorrow, see if we can drive it out.

**Comment by hblixt** [comment]
Is the idea that this would get the user on a guided path with examples, best practices and automated configuration? Probably would make sense to break it up in phases, but I like the idea. 
Having a day0 or day1 selection that will automatically install and configure the product and guide the user to get over the initial hurdle of getting started makes a lot of sense.

**Comment by edlee2121** [comment]
@hblixt also to get input directly from the user rather than the person who installed it or happens to come to our slack channels

**Comment by alexec** [comment]
Pending text from @hblixt  for survey privacy policy.

**Comment by alexec** [comment]
@hblixt can I bump this?

File: docs/use-cases/other.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Other
+
+## Docs
+
+We don't have any special docs for "other". Try these:
```
Comment (terrytangyuan) [inline docs/use-cases/other.md:5]: Perhaps also link to https://github.com/terrytangyuan/awesome-argo?

File: docs/use-cases/machine-learning.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Machine Learning
+
+## Docs
+
+* [Quick start](../quick-start.md) and [training](../training.md)
+* Try out the updated [Python and Java SDKs](https://github.com/argoproj/argo-workflows/tree/master/sdks).
+* [Authoring and Submitting Argo Workflows using Python](https://blog.argoproj.io/authoring-and-submitting-argo-workflows-using-python-aff9a070d95f)
+* Head to the [Kubeflow docs](https://www.kubeflow.org/docs/components/pipelines/).
+
+## Videos
```
Comment (terrytangyuan) [inline docs/use-cases/machine-learning.md:10]: There are several new ones for ML in https://github.com/terrytangyuan/awesome-argo#videos. Search for "Machine Learning" there.

File: docs/use-cases/other.md (hunk 2 above)
Comment (alexec) [reply docs/use-cases/other.md:5]: Maybe in a subsequent PR? Would you like to add? This is all open for contribution.

Comment (terrytangyuan) [reply docs/use-cases/other.md:5]: Sure

**Comment by alexec** [comment]
@terrytangyuan are you happy to approve this PR?

**Review by terrytangyuan — APPROVED** [review]
Yes the changes look great!

**Pull request merged**